_posts/alex_lepsa/2018-03-22-Spring-Method-Security-Using-JWTs.md

```diff
@@ -0,0 +1,342 @@
+---
+layout: post
+title:  "Using JWT's with Spring Security's @PreAuthorize annotation for method specific security"
+breadcrumb: true
+author: alex_lepsa
+date: 2018-03-22
+categories: alex_lepsa
+tags:
+    - Spring Security
+    - JWT
+teaser:
+    info: We explore how to implement a spring security strategy that statelessly authorizes a user using via JWT, allowing for method level permissions using the @PreAuthorize annotation.
+    image: alex_lepsa/2018-03/spring_jwt_thumb.jpg
+header:
+    version: small
+    title: Software Engineering Blog
+    image: header-logo-crop.png
+    icon: icon-blog 
+---
+
+Much has been written about integrating JWT's into spring security, and in fact Pivotal has included more and more first-class support for JWT's in their recent releases. That said, one thing that seems to be missing is a summary on how to stitch JWT's into an existing application using the `@PreAuthorize` annotation for fine-grain access control. We will be referencing our [SONG](https://www.overture.bio/song) application from [Overture](https://www.overture.bio/).
```

Comment:
In this blog post, we will be referencing...

Reply:
Yes, good suggestion, thanks!